### PR TITLE
Add note about contentfulness of input elements

### PR DESCRIPTION
**Pull request by noamr**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/paint-timing/pull/73.html" title="Last updated on Mar 23, 2020, 4:28 PM UTC (7644006)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/paint-timing/73/92f4392...7644006.html" title="Last updated on Mar 23, 2020, 4:28 PM UTC (7644006)">Diff</a>

### CODE REVIEW

File: painttiming.bs [hunk 1]
```diff
@@ -123,6 +128,7 @@ An [=element=] |target| is <dfn export>contentful</dfn> when one or more of the
 * |target| is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
 * |target| is a [=video element=] that [=represents=] its [=poster frame=] or the first video frame and the frame is available.
 * |target| is an [=svg element with rendered descendants=].
+* |target| is an [=input=] element, with its [=type attribute=] in any state other than [=hidden=].
```
Comment (rniwa) [inline painttiming.bs]: How about textarea?

Comment (rniwa) [reply painttiming.bs]: It seems a bit fragile to enumerate the list of elements & attributes like this. We also have to enumerate select, progress, meter, etc... I wonder if we can somehow the concept of replaced elements?

Comment (noamr) [reply painttiming.bs]: I think the most precise available word would be [represents](https://html.spec.whatwg.org/multipage/dom.html#represents). From [form elements](https://html.spec.whatwg.org/multipage/form-elements.html#the-datalist-element), "... the datalist element represents nothing ..."

Comment (noamr) [reply painttiming.bs]: OK I found a good definition: [being rendered](https://html.spec.whatwg.org/multipage/rendering.html#being-rendered). Added to "paintable", so:
* all form controls and inputs are "contentful"
* but only ones that are "being rendered" (generate boxes) are "paintable".

Comment (yoavweiss) [reply painttiming.bs]: SGTM

**Comment by npm1** [comment]
I don't think this is necessary. It's already covered by `|target| has a [=text node=] child, with at least one character, excluding [=document white space characters=]`.

**Comment by noamr** [comment]
> I don't think this is necessary. It's already covered by `|target| has a [=text node=] child, with at least one character, excluding [=document white space characters=]`.

buttons, meters, text inputs, empty text-area. all don't have a text-node child.

**Comment by npm1** [comment]
> > I don't think this is necessary. It's already covered by `|target| has a [=text node=] child, with at least one character, excluding [=document white space characters=]`.
> 
> buttons, meters, text inputs, empty text-area. all don't have a text-node child.

Should those be considered FCP though?

**Comment by npm1** [comment]
> I think at least a button or a text-input with a non-empty value should...
> It's debatable, I could go either way - but it should be defined.

What do you mean by nonempty value? Doesn't that mean having text, which again, would be covered already? If they don't count as contentful themselves, why do we need to explicitly state that in the definition of something that IS contentful?

**Comment by noamr** [comment]
> > I think at least a button or a text-input with a non-empty value should...
> > It's debatable, I could go either way - but it should be defined.
> 
> What do you mean by nonempty value? Doesn't that mean having text, which again, would be covered already? If they don't count as contentful themselves, why do we need to explicitly state that in the definition of something that IS contentful?

`<input type="text" value="abc" />` would not be covered by the current definition of text, as it doesn't generate text nodes.

**Comment by npm1** [comment]
Oh ok I thought it would generate text nodes. In that case yes this is needed :) In that case, the same clarification about nonempty text (excluding whitespace) would be needed.

File: painttiming.bs (hunk 1 above)
Comment (noamr) [reply painttiming.bs]: Revised. Also, there's no need to define `textarea` here specifically, as `textarea` generates text-nodes, so a non-empty `textarea` would already be considered contentful.

File: painttiming.bs [hunk 2]
```diff
@@ -65,6 +66,12 @@ urlPrefix: https://www.w3.org/TR/css-images-3/
 urlPrefix: https://html.spec.whatwg.org/multipage/media.html
     type: dfn; text: poster frame; url: #poster-frame;
     type: dfn; text: video element; url: #the-video-element;
+urlPrefix: https://html.spec.whatwg.org/multipage/input.html
+    type: dfn; text: input; url: #the-input-element;
```
Comment (npm1) [inline painttiming.bs]: I wonder if these are all indeed needed?

Comment (noamr) [reply painttiming.bs]: No, removed.

**Pull request merged**